src/main/java/duke/main/Duke.java

```diff
@@ -0,0 +1,160 @@
+package duke.main;
+
+import duke.exceptions.DukeException;
+import duke.exceptions.TaskException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+
+public class Duke {
+    private Storage storage;
+    private TaskList tasks;
+    private Parser parser;
+    private Ui ui;
+
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = storage.load();
+        } catch (FileNotFoundException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Adds a new task depending on type and append in to tasklist.
+     *
+     * @param whole_str the whole input string.
+     * @param str the whole input string, split.
+     * @throws TaskException
+     */
+    private void addTask(String whole_str, String[] str) throws TaskException {
+        if (str[0].compareTo("deadline") == 0) {
+            if (str.length == 1) {
+                throw new TaskException("deadline");
+            }
+            String stuff = "";
+            String deadline = "";
+
+            for (int i = 1; i < str.length; i++) {
+                if (str[i].compareTo("/by") == 0) {
+                    for (int j = 1; j < i; j++) {
+                        stuff += str[j] + " ";
+                    }
+                    for (int j = i + 1; j < str.length; j++) {
+                        deadline += str[j] + " ";
+                    }
+                    break;
+                }
+            }
+            tasks.add(new Deadline(stuff, parser.parseDate(deadline)));
+        } else if (str[0].compareTo("todo") == 0) {
+            if (str.length == 1) {
+                throw new TaskException("todo");
+            }
+
+            String stuff = "";
+            for (int i = 1; i < str.length; i++) {
+                stuff += str[i] + " ";
+            }
+            tasks.add(new Todo(stuff));
+        } else if (str[0].compareTo("event") == 0) {
+            if (str.length == 1) {
+                throw new TaskException("event");
+            }
+
+            String stuff = "";
+            String deadline = "";
+
+            for (int i =1;i< str.length;i++) {
+                if (str[i].compareTo("/at") == 0) {
+                    for (int j =1;j<i;j++) {
+                        stuff += str[j] + " ";
+                    }
+                    for (int j =i+1;j<str.length;j++) {
+                        deadline += str[j] + " ";
+                    }
+                    break;
+                }
```

Comment:
A neat set of for-loops!

However, just a suggestion:
```suggestion
for (int j  = 1; j < i; j++) {
    stuff += str[j] + " ";
}
for (int j = i + 1; j < str.length; j++) {
    deadline += str[j] + " ";
}
break;